solr/core/src/java/org/apache/solr/schema/BoolField.java

```diff
@@ -211,7 +212,7 @@ public List<IndexableField> createFields(SchemaField field, Object value) {
   @Override
   public Object toNativeType(Object val) {
     if (val instanceof CharSequence) {
-      return Boolean.valueOf(val.toString());
+      return Boolean.valueOf(StrUtils.parseBoolean(val.toString()));
```

Comment:
I think you could easily broaden the parameter type of parseBoolean to be CharSequence; then you're not calling val.toString.

Reply:
Thanks for pointing out the change in toInternal(). Makes sense. 

About the parameter type change in StrUtils.parseBoolean() to CharSeqeunce, is there a performance gain to be achieved there? With the proposed change, I'd have to typecast val to CharSequence while calling parseBoolean() inside toNativeType(). Alterntively, calling val.toString() should not have any overhead either since val would already be a String and the String class toString() implementation returns "this" anyway, so there is no extra object creation. We can avoid a change to StrUtils unless I am missing something. Thoughts welcome.

Reply:
if "val" is not a String, then there is an avoidable String conversion.  Any way, unless someone bothers to benchmark this (which would be a very dubious use of anyone's time), we don't know for sure, so feel free to ignore my proposal.

Reply:
Although I couldn't find evidence that val could be any other implementation of CharSequence, I agree with you it's one of those things not worth splitting hairs on. And it is definitely worth it being 100% sure rather than 99.99% in the interest of everyone's time :) Incorporating the comment.  Thanks.